core/blockchain.go

```diff
@@ -3015,3 +3034,155 @@ func EnablePersistDiff(limit uint64) BlockChainOption {
 		return chain
 	}
 }
+
+func (bc *BlockChain) GetRootByDiffHash(blockNumber uint64, blockHash common.Hash, diffHash common.Hash) (*types.VerifyResult, error) {
```

Comment:
I am not sure wether this func should return `error`, because if return error, the connection with peers will close, is it what we expected?

Reply:
Yes, since the fast node receive error msg, it will add the verify node into bad peer, this connection is expected to close.

Reply:
As we discussed, bad peer is within task, so no need to close the connection.  Please do not return error here.

Reply:
Updated.